### PR TITLE
Add patch support for pymysql (pure Python driver)

### PR DESCRIPTION
**Pull request by anil-grexit**
*Issue #214*

*Description of changes:*
Added patch support for `pymysql` client library (written in python) for MySQL.

TODO:
- [x] Add `unpatch` method
- [x] Add tests (Use: [this](https://pypi.org/project/testing.mysqld/))

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: tests/ext/pymysql/test_pymysql.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+                              port=dsn['port'])
+        cur = conn.cursor()
+        cur.execute(q)
+
+    subsegment = xray_recorder.current_segment().subsegments[-1]
+    assert subsegment.name == 'execute'
+    sql = subsegment.sql
+    assert sql['database_type'] == 'MySQL'
+    assert sql['user'] == dsn['user']
+    assert sql['database_version']
```
Comment (srprash) [inline tests/ext/pymysql/test_pymysql.py:50]: We should also test sql["driver_version"] == "PyMySql"

File: tests/ext/pymysql/test_pymysql.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+            cur.execute(q)
+        except Exception:
+            pass
+
+    subsegment = xray_recorder.current_segment().subsegments[-1]
+    assert subsegment.name == "execute"
+    sql = subsegment.sql
+    assert sql['database_type'] == 'MySQL'
+    assert sql['user'] == dsn['user']
+    assert sql['database_version']
```
Comment (srprash) [inline tests/ext/pymysql/test_pymysql.py:74]: We should also test sql["driver_version"] == "PyMySql"

File: tests/ext/pymysql/test_pymysql.py (hunk 1 above)
Comment (anil-grexit) [reply tests/ext/pymysql/test_pymysql.py:51]: Added

**Comment by anil-grexit** [comment]
@srprash requested changes done. Thanks for picking this up so quickly.

**Pull request merged**